guides/troubleshooting/inotify-watch-limits.md

```diff
@@ -0,0 +1,275 @@
+---
+title: inotify Watcher Limit Problems
+description: Learn how to resolve problems related to the inotify Watcher Limit.
+---
+
+When using some applications and tools, including Webpack or [code-server], you
+may encounter an error similar to the following:
+
+> Watchpack Error (watcher): Error: ENOSPC: System limit for number of file
+> watchers reached, watch '/some/path'
+
+[code-server]: https://github.com/cdr/code-server
+
+This article will show you how to diagnose and troubleshoot this error, which
+relates to an elevated number of inotify watchers in use.
+
+## Background
+
+[`Inotify`] allows programs to monitor files for changes, so that they receive
+an event whenever a user or program modifies a file. `inotify` requires kernel
+resources (memory and processor) for each file it tracks. As a result, the Linux
+kernel limits the number of file watchers that each user can register. The
+default settings vary according to the host system distribution; on Ubuntu 20.04
+LTS, the default limit is 8,192 watches per instance.
+
+[`Inotify`]: https://en.wikipedia.org/wiki/Inotify
+
+On a 64-bit system, each `inotify` watch that programs register will consume
+~1 kB of kernel memory, which cannot be swapped to disk and is not
+counted against the environment memory limit setting.
+
+## Diagnosis
+
+If you encounter the error that's the focus of this article, the total number of
+watchers in use is approaching the `max_user_watches` setting. The following
+sections will show you how to verify if this is the case.
+
+### Check tunable settings
+
+There are three kernel tuning options related to the `inotify` system:
+
+- `fs.inotify.max_queued_events`: The upper bound on the number of file
+  notification events pending delivery to programs
+- `fs.inotify.max_user_instances`: The maximum number of `inotify` instances per
+  user. Programs using `inotify` will typically create a single _instance_, so
+  this limit is unlikely to cause issues
```

Comment:
since this is two sentences, I think we should end the bullet with a full stop (and then the other bullets as well, for consistency) -- or we should revise the sentence to avoid the full stop if we want to avoid it